airflow/dag_processing/manager.py

```diff
@@ -1207,6 +1209,7 @@ def _kill_timed_out_processors(self):
                     processor.start_time.isoformat(),
                 )
                 Stats.decr("dag_processing.processes", tags={"file_path": file_path, "action": "timeout"})
+                Stats.gauge("dag_processing.processes_count", -1, delta=True, tags={"file_path": file_path, "action": "timeout"})
                 Stats.incr("dag_processing.processor_timeouts", tags={"file_path": file_path})
```

Comment:
@potiuk @hussein-awala @ferruzzi  I could see from the docs that the metric "dag_processing.processor_timeouts" is specified as gauge. But in the code, it is getting incremented as counter type here. Is this a bug?

Reply:
Not a bug as far as I am concerned.  StatsD does not have the UpDownCounter concept and their Gauges can be incremented/decremented.  It's just an Otel vs StatsD semantics issue.  In your other comments below, you point out the issues with making it a Gauage pretty well and more or less explain the reasons I did it the way I did.

Reply:
@ferruzzi I could see that param "dag_processing.processor_timeouts" is only increased. I still didn't get the logic behind why it has to be incremented as counter but specified as gauge in the documentation. Can you help me understand it?

Reply:
`processor_timeouts` is a counter, and is listed as one in the docs on the website:
![image](https://github.com/apache/airflow/assets/1920178/5d5ae067-7bdc-4525-982d-e551ac9fed16)

The only UpDownCounter in use is `dag_processing.processes`, which the docs also show in the docs as a Counter.
![image](https://github.com/apache/airflow/assets/1920178/6029cbbb-de79-4874-8cd9-e60f62ce68f2)

Reply:
ah, my bad @ferruzzi. My brain tricked me when i was seeing the documentation on website and sorry for bothering

Reply:
It's not the most straight-forward thing, and the metric names are all so similar, I totally get it.    If you see a way to make it more clear, please do!  Documentation improvements and clarifications are often overlooked and at least as important as code changes.

Reply:
That said, where does that leave this PR?   Is it still needed? There have been a few metrics-related ones making a bit of a tangled mess lately and I'm a bit lost on what needs to be looked at and where things are with each.